src/luaclang.c

```diff
@@ -5,50 +5,68 @@
+        CXIndex idx;
+        CXTranslationUnit tu;
+} clang_object;
 
-static CXIndex *new_CXIndex(lua_State *L) 
+/* Create clang_object as userdata */
+static clang_object *new_clangobject(lua_State *L) 
 {
-        CXIndex *idx = (CXIndex*) lua_newuserdata(L, sizeof(CXIndex));
-        luaL_getmetatable(L, INDEX_METATABLE);
+        clang_object *clang_parser = (clang_object*) lua_newuserdata(L, sizeof(clang_object));
+        luaL_getmetatable(L, CLANG_METATABLE);
         lua_setmetatable(L, -2);
-        return idx;
+        return clang_parser;
 }
 
-/* Convert userdata type to CXIndex */
-
-static CXIndex to_CXIndex(lua_State *L, int n)
+/* Convert userdata type to  */
+static clang_object to_clangobject(lua_State *L, int n) 
 {
-        CXIndex *idx = (CXIndex*) luaL_checkudata(L, n, INDEX_METATABLE);
-        return *idx;
+        clang_object *clang_parser = (clang_object*) luaL_checkudata(L, n, CLANG_METATABLE);
+        return *clang_parser;
 }
 
-/* Create CXTranslatioUnit as userdata */
-
-static CXTranslationUnit *new_CXTU(lua_State *L)
+/*      
+        Format - newParser(file_name)
+        Parameter - file_name - The name of the source file to load 
+        More info - 1. https://clang.llvm.org/doxygen/group__CINDEX.html#ga51eb9b38c18743bf2d824c6230e61f93
+                    2. https://clang.llvm.org/doxygen/group__CINDEX__TRANSLATION__UNIT.html#ga2baf83f8c3299788234c8bce55e4472e
+        Returns clang object whose translation unit cursor can be obtained.
+*/
+static int create_clangobject(lua_State *L)
 {
-        CXTranslationUnit *tu = (CXTranslationUnit*) lua_newuserdata(L, sizeof(CXTranslationUnit));
-        luaL_getmetatable(L, TU_METATABLE);
-        lua_setmetatable(L, -2);
-        return tu;
+        const char *file_name = lua_tostring(L, 1);
+        clang_object *clang_parser = new_clangobject(L);
+        clang_parser->idx = clang_createIndex(1, 0);
+        const char *args[] = {file_name};
+        clang_parser->tu = clang_parseTranslationUnit(clang_parser->idx, 0, args, 1, 0, 0, CXTranslationUnit_None);
+        return 1;
 }
 
-/* Convert userdata type to CXTranslatioUnit */
-
-static CXTranslationUnit to_CXTU(lua_State *L, int n) 
+/*      
+        Format - parserdisposeParser()
+        Parameter - parser - Clang object to be disposed
+        More info - 1. https://clang.llvm.org/doxygen/group__CINDEX.html#ga51eb9b38c18743bf2d824c6230e61f93
+                    2. https://clang.llvm.org/doxygen/group__CINDEX__TRANSLATION__UNIT.html#ga2baf83f8c3299788234c8bce55e4472e
+        Returns nothing
+*/
+static int dispose_clangobject(lua_State *L)
 {
-        CXTranslationUnit *tu = (CXTranslationUnit*) luaL_checkudata(L, n, TU_METATABLE);
-        return *tu;
+        clang_object clang_parser = to_clangobject(L, 1);
+        if (clang_parser.idx == NULL) return 0;
+        clang_disposeIndex(clang_parser.idx);
+        clang_disposeTranslationUnit(clang_parser.tu);
+        clang_parser.idx = NULL;
```

Comment:
This assignment only modifies the local variable; it doesn't modify the userdata that Lua has stored.